src/usr/local/www/diag_reboot.php

```diff
@@ -49,6 +49,17 @@
     ($_POST['override'] != "yes"))):
 	if (DEBUG) {
 		print_info_box(gettext("Not actually rebooting (DEBUG is set true)."), 'success');
+	} elseif ($_POST['Submit'] == 'FSCK Reboot') {
+		mwexec('/sbin/nextboot -e "pfsense.fsck.force=5"');
+		print('<div><pre>');
```

Comment:
There is quite a lot of repeated code here in each section that could be simplified.

A switch statement should be used instead of a long chain of if/elseif/else